### PR TITLE
Feat: added periodic saves of config.json

### PR DESCRIPTION
**Pull request by TheRealGioviok**
So that no one must suffer anymore.

### CODE REVIEW

**Comment by gahtan-syarif** [comment]
and for this one:
```
long unsigned int save_iter = 0;
```
i think that should be:
```
long unsigned int save_iter = initial_matchcount_;
```

**Comment by Disservin** [comment]
please use the fixed size integer variants, std::uint32_t etc.

**Comment by TheRealGioviok** [comment]
On it.

**Comment by TheRealGioviok** [comment]
Should be working fine now. Tested it and it works. I also tried to follow the existing conventions for structs and so on.

**Comment by TheRealGioviok** [comment]
All this well written repo is getting to me 😵 . Do I just run make update-man and commit the stuff?

**Comment by gahtan-syarif** [comment]
and also for this:
```
Enables automatic save of the sprt state in the config.json file each N games played.
```
i suggest changing to
```
Enables automatic save of the tournament state in the config.json file each N games played.
```

**Comment by gahtan-syarif** [comment]
> All this well written repo is getting to me 😵 . Do I just run make update-man and commit the stuff?

yeah, run make update-man locally in your computer and it would change the man.hpp file, then you replace the old man.hpp with the new one in your repo

**Comment by gahtan-syarif** [comment]
> > All this well written repo is getting to me 😵 . Do I just run make update-man and commit the stuff?
> 
> yeah, run make update-man locally in your computer and it would change the man.hpp file, then you replace the old man.hpp with the new one in your repo

this is so that the new version would output the new man file when we run -help

**Comment by TheRealGioviok** [comment]
immagine

Success! Any other things to fix?

**Comment by gahtan-syarif** [comment]
> immagine
> 
> Success! Any other things to fix?

looks good to me, we'll wait for disservin

File: src/matchmaking/tournament/base/tournament.hpp [hunk 1]
```diff
@@ -24,7 +24,10 @@ class BaseTournament {
     BaseTournament(const options::Tournament &config,
                    const std::vector<EngineConfiguration> &engine_configs);
 
-    virtual ~BaseTournament() = default;
+    ~BaseTournament() {
```
Comment (Disservin) [inline src/matchmaking/tournament/base/tournament.hpp:27]: idk prefer if you kept `virtual` here

**Review by Disservin — CHANGES_REQUESTED** [review]

**Pull request merged**